src/wallet/wallet.cpp

```diff
@@ -3302,23 +3302,21 @@ bool ReserveDestination::GetReservedDestination(CTxDestination& dest, bool inter
     if (nIndex == -1)
     {
         CKeyPool keypool;
-        if (!m_spk_man->GetReservedDestination(type, internal, nIndex, keypool)) {
+        if (!m_spk_man->GetReservedDestination(type, internal, dest, nIndex, keypool)) {
             return false;
         }
         vchPubKey = keypool.vchPubKey;
         fInternal = keypool.fInternal;
+        address = dest;
     }
```

Comment:
In commit "Key pool: Move LearnRelated and GetDestination calls" (393315354da454eb0f9322b92ad736bd23606f97

Do we need:

```diff
@@ -3308,6 +3308,8 @@ bool ReserveDestination::GetReservedDestination(CTxDestination& dest, bool inter
         vchPubKey = keypool.vchPubKey;
         fInternal = keypool.fInternal;
         address = dest;
+    } else {
+        dest = address;
     }
     assert(vchPubKey.IsValid());
     return true;
```

It seems like `dest` is currently left unset in the `nIndex != -1` case where previously it was always set.

Reply:
Done as suggested.

Reply:
Oops, moved it to the right commit.